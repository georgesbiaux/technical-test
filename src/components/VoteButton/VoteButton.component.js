import React from 'react';
import StyledVoteButton from './VoteButton.style';

const VoteButton = props => (
  <StyledVoteButton className="vote-button" userVoted={props.userVoted}>
    {props.votes}
    <i className="fa fa-heart" />
  </StyledVoteButton>
);

export default VoteButton;
