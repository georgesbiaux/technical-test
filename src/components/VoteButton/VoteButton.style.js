import styled from 'styled-components';

const StyledButton = styled.button`
  border: 0;
  font-size: 22px;
  width: 70px;
  height: 40px;
  border-radius: 20px;
  background-color: ${props => (props.userVoted ? '#d06769' : '#343434')};
  color: #ffffff;
  cursor: pointer;
  margin: 5px;
  transition: background-color 0.2s ease;
  font-weight: 200;

  :focus {
    outline: -webkit-focus-ring-color auto 0px;
  }

  :hover {
    background-color: ${props => (props.userVoted ? '#b2615f' : '#343434')};
  }

  i {
    margin-left: 5px;
  }
`;

export default StyledButton;
