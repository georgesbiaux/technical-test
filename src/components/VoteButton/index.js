export { default } from './VoteButton.component';
