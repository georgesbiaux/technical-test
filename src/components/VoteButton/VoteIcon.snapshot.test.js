import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import VoteButton from './VoteButton.component';

describe('<VoteButton>', () => {
  it('should render correctly when userVoted is false', () => {
    const props = {
      userVoted: false,
      votes: 0,
    };
    const tree = render(<VoteButton {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('should render correctly when userVoted is true', () => {
    const props = {
      userVoted: true,
      votes: 3,
    };
    const tree = render(<VoteButton {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
