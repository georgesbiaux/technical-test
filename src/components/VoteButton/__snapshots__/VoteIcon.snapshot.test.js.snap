// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VoteButton> should render correctly when userVoted is false 1`] = `
<button
  class="vote-button sc-bdVaJa QUdHO"
>
  0
  <i
    class="fa fa-heart"
  >
    
  </i>
</button>
`;

exports[`<VoteButton> should render correctly when userVoted is true 1`] = `
<button
  class="vote-button sc-bdVaJa fMBtmC"
>
  3
  <i
    class="fa fa-heart"
  >
    
  </i>
</button>
`;
